tensorflow/compiler/xla/python/pytree.cc

```diff
@@ -90,27 +90,50 @@ namespace py = pybind11;
   return it == registry->registrations_.end() ? nullptr : it->second.get();
 }
 
-/*static*/ std::vector<py::object> GetSortedPyDictKeys(PyObject* py_dict) {
-  std::vector<py::object> keys;
-  keys.reserve(PyDict_Size(py_dict));
-  PyObject* key;
-  Py_ssize_t pos = 0;
-  while (PyDict_Next(py_dict, &pos, &key, /*value=*/nullptr)) {
-    keys.push_back(py::reinterpret_borrow<py::object>(key));
-  }
-
-  int ret = 0;
-  std::stable_sort(keys.begin(), keys.end(),
-                   [&ret](const py::object& a, const py::object& b) {
-                     int cmp =
-                         PyObject_RichCompareBool(a.ptr(), b.ptr(), Py_LT);
-                     if (cmp == -1) ret = -1;
-                     return cmp;
-                   });
-  if (ret == -1) {
-    throw py::error_already_set();
-  }
```

Comment:
The old implementation is `py::list -> std::vector -> std::sort`. Changed to `py::list -> py::sort -> std::vector` for better Python exception handling.